pkg/core/http.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2022 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/zapr"
+	"go.uber.org/zap"
+
+	"github.com/chaos-mesh/chaos-mesh/pkg/chaosdaemon/tproxyconfig"
+	"github.com/pingcap/errors"
+)
+
+const (
+	TargetRequest  tproxyconfig.PodHttpChaosTarget = "Request"
+	TargetResponse tproxyconfig.PodHttpChaosTarget = "Response"
+)
+
+const (
+	HTTPAbortAction  = "abort"
+	HTTPDelayAction  = "delay"
+	HTTPConfigAction = "config"
+)
+
+var _ AttackConfig = &HTTPAttackConfig{}
+
+type HTTPAttackConfig struct {
+	CommonAttackConfig
+	Config   tproxyconfig.Config
+	ProxyPID int
+
+	Logger logr.Logger
+}
+
+func (c HTTPAttackConfig) RecoverData() string {
+	data, _ := json.Marshal(c)
+	return string(data)
+}
+
+type HTTPAttackOption struct {
+	CommonAttackConfig
+	ProxyPorts []uint                            `json:"proxy_ports,omitempty"`
+	Rule       tproxyconfig.PodHttpChaosBaseRule `json:"rule"`
```

Comment:
How about using a new type to register options? I think we do not need the whole `PodHttpChaosBaseRule`, the several fields can meet our requirements. And more importantly, the nullable field is not suitable for command-line options.

Reply:
I agree with you.

Reply:
Fixed.